packages/vuetify/src/labs/VNumberInput/__tests__/VNumberInput.spec.cy.tsx

```diff
@@ -158,4 +158,97 @@ describe('VNumberInput', () => {
         .then(() => expect(numberInputValue.value).to.equal(0))
     })
   })
+
+  describe('parsing pasted values', () => {
+    it('should parse valid numbers correctly', () => {
+      const scenarios = [
+        { text: '-0', expected: '0' },
+        { text: '-00123', expected: '-123' },
+        { text: '.250', expected: '0.25' },
+        { text: '000.321', expected: '0.321' },
+        { text: '100.000', expected: '100' },
+      ]
+      scenarios.forEach(({ text, expected }) => {
+        cy.mount(() => <VNumberInput />)
+        cy.window().its('navigator.clipboard').then((clip: Clipboard) => clip.writeText(text))
+        cy.get('.v-number-input input').focus()
+        cy.document().invoke('execCommand', 'paste')
+        cy.get('.v-number-input input').blur().should('have.value', expected)
+      })
+    })
+
+    it('should retrieve digits when pasted value contains some mixed characters', () => {
+      const scenarios = [
+        { text: ' 1,250.32\n', expected: '1250.32' },
+        { text: 'text 1024 text', expected: '1024' },
+        { text: '- 1123.', expected: '-1123' },
+      ]
+      scenarios.forEach(({ text, expected }) => {
+        cy.mount(() => <VNumberInput />)
+        cy.window().its('navigator.clipboard').then((clip: Clipboard) => clip.writeText(text))
+        cy.get('.v-number-input input').focus()
+        cy.document().invoke('execCommand', 'paste')
+        cy.get('.v-number-input input').blur().should('have.value', expected)
+      })
+    })
+
+    it('should use as much characters as possible to display value containing only received digits', () => {
+      const scenarios = [
+        { text: '0.000010', expected: '0.00001' },
+        { text: '0.012340', expected: '0.01234' },
+        { text: '099999990.000000010', expected: '99999990.00000001' },
+        { text: '99999999999999999999999999999', expected: '999999999999999' },
+        { text: '0.999999999999999999999999999', expected: '0.999999999999999' },
+        { text: '3.9999999999999995', expected: '3.999999999999999' },
+      ]
+      scenarios.forEach(({ text, expected }) => {
+        cy.mount(() => <VNumberInput />)
+        cy.window().its('navigator.clipboard').then((clip: Clipboard) => clip.writeText(text))
+        cy.get('.v-number-input input').focus()
+        cy.document().invoke('execCommand', 'paste')
+        cy.get('.v-number-input input').blur().should('have.value', expected)
+      })
+    })
+  })
+
+  describe('typing values', () => {
+    it('should ignore invalid and duplicated characters', () => {
+      const scenarios = [
+        { text: '+=1234... abc e12', expected: '1234.12' },
```

Comment:
Invalid input should be enforced to empty `undefined`

Reply:
I wanted a test case for number extraction, but this example does not convey that we wish to accept numbers copy/pasted with any format (as long as decimal separator is dot). I will update this part.